DataFormats/PatCandidates/src/classes_def_other.xml

```diff
@@ -71,6 +71,23 @@
   <class name="edm::Wrapper<edm::ValueMap<vid::CutFlowResult> >"/>
   <class name="pat::UserHolder<vid::CutFlowResult>"/>
 
+  <class name="pat::HcalDepthEnergyFractions" ClassVersion="3">
+    <version ClassVersion="3" checksum="4055779358"/>
+    <field name="fractions_" transient="true"/>
+  </class>
+  <class name="std::vector<pat::HcalDepthEnergyFractions>"/>
+  <class name="edm::Wrapper<std::vector<pat::HcalDepthEnergyFractions> >"/>
+  <class name="edm::ValueMap<pat::HcalDepthEnergyFractions>" />
+  <class name="edm::Wrapper<edm::ValueMap<pat::HcalDepthEnergyFractions> >"/>
+
+  <ioread sourceClass = "pat::HcalDepthEnergyFractions" version="[1-]"
+                targetClass="pat::HcalDepthEnergyFractions"
+		source="std::vector<uint8_t> fractionsI_"
+		target="fractions_">
+  <![CDATA[ newObj->initFloatVector();
+  ]]>
```

Comment:
Static analyzer picked up on this line
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26465/34296/llvm-analysis/report-9a8c17.html#EndPath

According to it, the code for this rule looks like the following

```c++
 pat::HcalDepthEnergyFractions* newObj = (pat::HcalDepthEnergyFractions*)target;

 if (oldObj) {}

 if (newObj) {}

  newObj->initFloatVector();

```

@pcanal @Dr15Jones 
By some magic (?) I would expect that the body of ```CDATA[``` should end up inside ```if (newObj) {```.
Are we using the syntax for ioread incorrectly or should we simply add ```if (newObj) {``` ?

We have the same pattern in a few other ioread rules in CMSSW. So, a more general solution is perhaps outside the scope of this PR.

Reply:
That code is there solely to prevent unused variable warning.  It needs to be updated to be ```(void)newObj```, etc.